### PR TITLE
#4500 - fixed radio buttons not selecting correctly

### PR DESCRIPTION
**Pull request by florianf22**
**Related issue(s):**
* Fixes https://github.com/scandipwa/scandipwa/issues/4500

**Problem:**
* Radio buttons weren't behaving correctly due to e.preventDefault();

**In this PR:**
* removed e.preventDefault();

### CODE REVIEW

**Comment by florianf22** [comment]
Hi, I was almost sure it was due to preventDefault, but couldn't find anything. So, I decided deploy unchanged and it works, no weird behaviours

**Comment by florianf22** [comment]
Hi, I haven't changed the code in any way. that issues had been worked on and has already been merged, as I can see. if the issue still arises, I suggest to open another PR, this branch isn't suited for that

**Comment by aleksandrakorolova** [comment]
Please, merge master in this PR

**Comment by florianf22** [comment]
@carinadues could you please take a look at this?

**Comment by florianf22** [comment]
My master branch wasn't up to date, made changes, it should work now.

File: packages/scandipwa/src/component/Field/Field.component.js [hunk 1]
```diff
@@ -193,21 +193,21 @@ export class Field extends PureComponent {
         const {
             type,
             setRef,
-            attr: { defaultChecked = false, ...newAttr } = {},
+            attr: { ...newAttr } = {},
```
Comment (zans-laksa) [inline packages/scandipwa/src/component/Field/Field.component.js:196]: The use of the rest operator is irrelevant if you don't get other values first.
Remove the `newAttr` and use `attr` instead.

File: packages/scandipwa/src/component/Field/Field.component.js [hunk 2]
```diff
@@ -193,21 +193,21 @@ export class Field extends PureComponent {
...
+        const { id = '', value = '' } = newAttr;
         const elem = type.charAt(0).toUpperCase() + type.slice(1);
         const inputEvents = {
             ...events,
             onChange: onChange || noopFn
         };
         // if button value is "none" do not disable
         const isButtonDisabled = (!value.match('none') && isDisabled);
-        const isChecked = isButtonDisabled || defaultChecked ? !isDisabled : null;
+        // const isChecked = isButtonDisabled || defaultChecked ? !isDisabled : null;
```
Comment (zans-laksa) [inline packages/scandipwa/src/component/Field/Field.component.js:210]: Please don't leave the commented code.

**Review by zans-laksa — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/component/Field/Field.component.js (hunk 2 above)
Comment (florianf22) [reply packages/scandipwa/src/component/Field/Field.component.js:210]: that comments were there already . they have now been removed

File: packages/scandipwa/src/component/Field/Field.component.js (hunk 1 above)
Comment (florianf22) [reply packages/scandipwa/src/component/Field/Field.component.js]: this was there already, removed spread, renamed the variable on the fly

Comment (zans-laksa) [reply packages/scandipwa/src/component/Field/Field.component.js]: Please don't rename variables without a reason. Use `attr` instead.

File: packages/scandipwa/src/component/Field/Field.component.js [hunk 3]
```diff
@@ -193,21 +193,20 @@ export class Field extends PureComponent {
...
             setRef,
-            attr: { defaultChecked = false, ...newAttr } = {},
+            attr: newAttr = {},
             events: { onChange },
             events,
             isDisabled,
             label
         } = this.props;
-        const { id = '', checked, value = '' } = newAttr;
+        const { id = '', value = '' } = newAttr;
```
Comment (zans-laksa) [inline packages/scandipwa/src/component/Field/Field.component.js:202]: Please don't rename variables without a reason. Use `attr` instead.

**Review by zans-laksa — CHANGES_REQUESTED** [review]

**Comment by aleksandrakorolova** [comment]
1. By default selected out of stock product => User can't add product to cart
Screenshot 2022-03-31 at 14 31 52

**Comment by florianf22** [comment]
All good now.

**Review by zans-laksa — APPROVED** [review]

**Comment by aleksandrakorolova** [comment]
1. Fixed 
2. Can reproduce issue with checkboxes from task (Can't unselect by defaut selected checkbox)
https://drive.google.com/file/d/1R23L0dmpe5yYOGZd70bXfEL8-UuEYHI2/view?usp=sharing

**Comment by florianf22** [comment]
I can't reproduce that. Please clear cache, it's working from what i see

**Comment by florianf22** [comment]
please check again

**Review by zans-laksa — APPROVED** [review]

**Comment by aleksandrakorolova** [comment]
1. Issue appears again - By default selected out of stock product => User can't add product to cart
Screenshot 2022-04-04 at 10 56 48
2. Fixed

**Review by zans-laksa — APPROVED** [review]

**Comment by aleksandrakorolova** [comment]
Fixed on PDP
But still can reproduce: 

**on PLP:**
 Checkmark stays in checkbox when user unselects filter
 https://drive.google.com/file/d/1Wpb0FG_ZR24Bg2IFBenH8hPpJODNGOh3/view?usp=sharing

**Steps:**
1. Open PLP
2. Select filter 
3. Unselect

**in My account:**
Can't unselect checkboxes 'This is default Shipping Address' and 'This is default Billing Address' in Edit address form in my account
https://drive.google.com/file/d/1TWktBeP6te3gkRhzh1_jrfaSyGcQFO3P/view?usp=sharing
**Steps:**
1. Open Sign in form
2. Log in
3. Open Address book
4. Click on 'Edit address' for default shipping and/or billing address
5. Unselect checkbox 'This is default Shipping Address' and/or 'This is default Billing Address'

**Comment by carinadues** [comment]
The issue will be fixed in https://github.com/scandipwa/scandipwa/pull/4701

**Pull request closed without merging**